Package.swift

```diff
@@ -594,11 +607,15 @@ let package = Package(
         "FirebaseABTesting",
         .product(name: "GULEnvironment", package: "GoogleUtilities"),
         .product(name: "nanopb", package: "nanopb"),
+        .target(name: "FirebaseInAppMessaging_iOS", condition: .when(platforms: [.iOS])),
+        .target(name: "FirebaseInAppMessaging_tvOS", condition: .when(platforms: [.tvOS])),
       ],
       path: "FirebaseInAppMessaging/Sources",
       exclude: [
         "DefaultUI/CHANGELOG.md",
         "DefaultUI/README.md",
+        "iOS",
```

Comment:
Not sure if it actually matters here but looks like in other target we add `/` to the directory names we are excluding, so we may need to change it to `"iOS/"`.

Reply:
Thanks for the tip! I can't tell if it makes a difference because the cause of the failures is elsewhere. I will keep this in mind though 🙂 

The failures we're now seeing are because I have added  platform-specific targets `FirebaseInAppMessaging_tvOS` and `FirebaseInAppMessaging_iOS`. AFAIU, while SPM supports targets having conditionally dependent platform-specific targets, it does not support platform-specific targets since all targets in the package manifest are compiled.

Currently, experimenting with rearranging our workflow tests to see if there is better way to test coverage. Will update soon!

Reply:
My guess was that exclude actually is not working here so the iOS and tvOS specific resource are still compiled for `FirebaseInAppMessaging` which triggers the compilation error. I think it's worth checking. But if the targets like `FirebaseInAppMessaging_tvOS` are still compiled for e.g. macOS even if they are not included as a dependency anywhere, then it's disappointing.

Reply:
I believe targets like `FirebaseInAppMessaging_tvOS` are still compiled for e.g. macOS even if they are not included as a dependency anywhere. Here's a screenshot of the compile failure. Despite building for iOS, the packahe manifest still tries to compile the `FirebaseInAppMessaging_tvOS` target.

<img width="887" alt="Screen Shot 2021-03-31 at 2 11 42 PM" src="https://user-images.githubusercontent.com/36927374/113201664-c459ad00-9237-11eb-892a-a6ecae8f11af.png">

Reply:
Disappointing... I would be curious if this an issue for the tests only or for a real project using Firebase as well.